File: 준영/124 나라의 숫자.js
function solution(n) {
  var answer = ''
  var role = ['4', '1', '2']
  var arr = []
  while (n) {
    var temp = n % 3
    arr.push(role[temp])
    n = parseInt(n / 3)
    if (temp == 0) {
      n -= 1
    }
  }
  arr.reverse()
  answer = arr.join('')
  return answer
}

File: 준영/프로그래머스/LV2/124 나라의 숫자.py
def solution(n):
    answer = ''
    role = ['4', '1', '2']
    arr = []
    while n:
        temp = n % 3
        arr.insert(0, role[temp])
        n //= 3 
        if temp == 0:
            n -= 1
    answer = ''.join(arr)
    return answer

# solution(1)
# solution(2)
solution(3)
# solution(4)
# solution(5)
# solution(6)
# solution(7)
# solution(8)
# solution(9)
# solution(10)